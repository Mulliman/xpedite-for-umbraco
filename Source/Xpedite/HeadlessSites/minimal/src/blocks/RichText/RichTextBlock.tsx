import { Block } from "@/umbraco/types";

type ContentProps = {
    richText: string
};

type Props = Block<ContentProps, any>;

export default function RichTextBlock({ content }: Props) {
    const richText = content?.properties?.richText || '';

    return (
        <>
            <div className='' dangerouslySetInnerHTML={{ __html: richText }}></div>
        </>
    )
}
