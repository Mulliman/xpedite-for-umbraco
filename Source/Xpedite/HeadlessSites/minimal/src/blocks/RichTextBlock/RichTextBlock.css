.block-rich-text-block {
    border: 2px solid #92C98C;
}

.block-rich-text-block-name {
    background: #92C98C;
    color: black;
}

.richText{
    max-width: 1200px;
    margin: 1rem auto;
}