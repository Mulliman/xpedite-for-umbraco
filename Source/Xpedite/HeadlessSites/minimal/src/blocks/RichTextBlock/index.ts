import RichTextBlock from "./RichTextBlock";

export default RichTextBlock;