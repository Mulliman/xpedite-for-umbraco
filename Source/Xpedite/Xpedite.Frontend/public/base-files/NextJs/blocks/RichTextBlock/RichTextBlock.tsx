import { Block } from "@/umbraco/types";

import "./RichTextBlock.css";

export type RichTextBlockContentProps = {
  richText?: any;
};

export type RichTextBlockSettingsProps = {};

type RichTextBlockProps = Block<RichTextBlockContentProps, RichTextBlockSettingsProps>;

export default function RichTextBlock({ content, settings }: RichTextBlockProps) {
  const { richText } = content?.properties || {};

  return (
    <div className="block-rich-text-block">
      <div className="block-rich-text-block-name">Block: RichTextBlock</div>

      <div className="richText" dangerouslySetInnerHTML={{ __html: richText?.markup || "" }}></div>
    </div>
  );
}
