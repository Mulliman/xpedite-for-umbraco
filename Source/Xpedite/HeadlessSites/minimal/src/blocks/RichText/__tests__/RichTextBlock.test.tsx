import React from 'react';
import { render } from '@testing-library/react';
import RichTextBlock from '../RichTextBlock';

describe('RichTextBlock', () => {

    it('matches snapshot', () => {
        const content = {
            contentType: 'test',
            id: '111-1111-1111',
            properties: {
                richText: '<p>This is a snapshot test</p>'
            },
            settings: {
                
            }
        };

        const { asFragment } = render(<RichTextBlock content={content} />);
        expect(asFragment()).toMatchSnapshot();
    });


    it('renders rich text content', () => {
        const content = {
            contentType: 'test',
            id: '111-1111-1111',
            properties: {
                richText: '<p>This is a test</p>'
            },
            settings: {
                
            }
        };

        const { container } = render(<RichTextBlock content={content} />);
        expect(container.innerHTML).toContain('<p>This is a test</p>');
    });

    it('renders empty div when richText is empty', () => {
        const content = {
            contentType: 'test',
            id: '111-1111-1111',
            properties: {
                richText: ''
            },
            settings: {
                
            }
        };

        const { container } = render(<RichTextBlock content={content} />);
        expect(container.innerHTML).toContain('<div class=""></div>');
    });
});