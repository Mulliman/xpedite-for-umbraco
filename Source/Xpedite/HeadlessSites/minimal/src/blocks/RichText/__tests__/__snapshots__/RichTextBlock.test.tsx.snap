// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichTextBlock matches snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <p>
      This is a snapshot test
    </p>
  </div>
</DocumentFragment>
`;
